linuxkit/linuxkit
# Update, fix (and extend a bit) Scaleway support

**- What I did**

* Updated scaleway-sdk to the latest tagged release and adjusted code appropriately.
* Fixed working with `ams1` zone.
* Added ability to specify resulting image size (defaulting to GB-rounded ISO size).
* Fixed some typos and such.

**- How I did it**

Vendored in new version, read sdk docs and changed code until it worked again.
Debugged `getImageiD` to find out that it did not differentiate images by zones (and tried to use one in `par1` for `ams1`, for example).
Generally looked over the docs and code as I was deploying my instances.

**- How to verify it**

Run for example `linuxkit push scaleway -zone ams1 -organization-id <ID> <ISO-FILE>`. Current master would fail with `http error 404 Not Found: "f974feac-abae-4365-b988-8ec7d1cec10d" not found`.
This patch should work as expected, plus also create image of size equal to \<ISO-FILE> size (rounded to GB, because SCW only operates on multiples of GB, it seems).
Size can also be added as argument with e.g. `-volume-size 2` (GB implied).

**- Description for the changelog**

* Scaleway: Fixed working in `ams1` zone.
* Scaleway: Added ability to specify resulting image volume size (with default of input ISO size rounded to Gigabytes).

Not sure if this should go to changelog or where, but there are few breaking changes here:
* Access Key is now mandatory. Not much that can be done really.
* Notions of `project` are no more. It is `organization` now. Can provide compat. flag, but opted out for now.

**- A picture of a cute animal (not mandatory but encouraged)**
![aveter_med](https://user-images.githubusercontent.com/431488/80765377-2cdd6d00-8b43-11ea-99ab-e2ee32c3ce91.png)
